### PR TITLE
Fixed a few spots where extendedType was misnamed as extendType

### LINKED ISSUE #7361: Property is named `extendType` when it should be `extendedType`
An object of type `QueryPropertyMetaData`, (specifically one returned from `ECSqlReader.getMetaData()`) according to [ConcurrentQuery.ts](https://github.com/iTwin/itwinjs-core/blob/d8b4364092ea7f00c0340179a51ff1ae126395f8/core/common/src/ConcurrentQuery.ts#L4), should have a property `extendType`. 

However, if you try to access it, it will be undefined:
Image
Image

Instead, the property can be accessed by calling `extendedType`:
Image
Image

### PR DESCRIPTION
**Pull request by DonataSnieskaite**
closes #7361 

~Fixed the above issue in [ConcurrentQuery.ts](https://github.com/iTwin/itwinjs-core/blob/d8b4364092ea7f00c0340179a51ff1ae126395f8/core/common/src/ConcurrentQuery.ts#L4) and a few other files.~

Changed the misnamed extendType to extendedType in a comment and documentation.

### CODE REVIEW

**Comment by hl662** [comment]
No tests needed to be updated?

**Comment by DonataSnieskaite** [comment]
> No tests needed to be updated?

Not sure, I searched for usage of "extendType" in the repo and found nothing (except for the few spots I replaced)

**Comment by hl662** [comment]
@aruniverse `QueryPropertyMetadata` interface is marked public, and this is changing the interface type. It's a bug fix though, so should we backport this to 4.10 + 4.11, or leave it till 5.0

File: core/ecschema-editing/src/Editing/Properties.ts
Comment (DonataSnieskaite) [inline core/ecschema-editing/src/Editing/Properties.ts:1]: This change only updates the comment to follow the already existing code (look at line 324, the parameter is already named `extendedTypeName`), so it doesn't affect anything.

**Comment by aruniverse** [comment]
have you seen https://github.com/iTwin/itwinjs-core/pull/7372 ?

File: common/api/core-common.api.md [hunk 1]
```diff
@@ -7513,7 +7513,7 @@ export enum QueryParamType {
 export interface QueryPropertyMetaData {
     accessString?: string;
     className: string;
-    extendType: string;
+    extendedType: string;
```
[the remark this answers is not in the document]
Comment (DonataSnieskaite) [reply common/api/core-common.api.md:7516]: I didn't consider people using this in other repos, which is a big oversight. I'll leave it up to Soham to fix this in his PR, and I'll later merge this for the remaining (comment) changes.

File: common/changes/@itwin/ecschema-editing/Donata-fix-extended-type_2024-11-18-14-30.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/ecschema-editing",
+      "comment": "renamed extendType to extendedType",
```
Comment (aruniverse) [inline common/changes/@itwin/ecschema-editing/Donata-fix-extended-type_2024-11-18-14-30.json:5]:
```suggestion
      "comment": "",
```

**Review by aruniverse — APPROVED** [review]

**Pull request merged**